### PR TITLE
Updated the `plugin` command to better handle paths

### PR DESCRIPTION
**Pull request by lethalbit**
This PR updates the `plugin -i` command to better deal with some edge cases and inconsistencies with how it behaved.

The first update is that it now ensures that the plugin does not end with `.so` before trying to append it. Previously when running `plugin -i nya.so` it would try to look for `nya.so.so` in the `YOSYS_SHARE/plugins` directory, that has been fixed.

The second update is that it now checks for the `.py` extension even without Python support being enabled, and produces an error message accordingly.

The final update is that the filename path is now also run through `rewrite_path`, as is done with a large portion of the other file inputs in Yosys.

This PR does not address #2545 nor #3151 yet, I wanted to get some basic restructuring done first to make implementation a little nicer.

### CODE REVIEW

**Comment by mmicko** [comment]
LGTM, but there are still situations where checking if plugin is already loaded could be problematic (already is like that so you did not broke anything), guess storing base name of file would be better than original filename.

**Comment by lethalbit** [comment]
@mmicko would you like me to implement the normalization for checking to see if the plugin is loaded or not?

My general idea for that is to just get the stem from the file name so if you load something like `+/plugins/foo.bar.so` it would just register it as `foo.bar.so`, although this could potentially cause problems if there are plugins with ambiguous stems. In which case some unique plugin identifier would be needed, or would that not be a problem?

**Comment by mmicko** [comment]
Hi @lethalbit, my plan is to merge this right after release (have to check with others but should be these days), just to have more time to test all in real use after merge, since now tools like eqy are relaying on plugin mechanism.

**Review by mmicko — APPROVED** [review]

**Comment by lethalbit** [comment]
Okay! Sounds good to me~

**Pull request merged**
